text/0143-sdk-fail-safe-mode.md

```diff
@@ -0,0 +1,512 @@
+
+1. When the SDK is in a critical state and potentially causing crashes, the last thing we want to do is a web request.
+2. It doesn't work for continuous SDK crashes during SDK init.
+3. It doesn't work offline.
+4. It requires manual action. We need to monitor our SDK crashes and input from customers continuously.
+5. It requires infrastructure changes.
+
+### Option B4: [Discarded] Bundling SDK versions at the same time  <a name="option-b4"></a>
+
+> 🚫 **Discard reason:** It has too many significant cons. 🚫
+
+The SDK ships with two different SDK versions. It has a wrapper for the user and then delegates the actual method calls to the duplicated SDK code. If the SDK detects it’s crashing often, it uses the fallback SDK version.
+
+No notes on [crashing scenarios](#crashing-scenarios), because we can discard this option as it has two many significant cons.
+
+#### Pros <a name="option-b4-pros"></a>
+
+1. When the SDK crashes, it can still function with the fallback SDK version.
+
+#### Cons <a name="option-b4-cons"></a>
+
+1. Roughly doubles the size of the SDK.
+2. It requires an extra package.
+3. Only a subset of customers might use this, and only high-quality aware customers might accept the tradeoff of a double-sized SDK. In fact, most high-quality aware customers most likely care about app size and will use the stable release channel.
+
+## C: Knowing When the SDK is Disabled <a name="c-knowing-when-the-sdk-is-disabled"></a>
+
+### Option C1: Failing SDK Endpoint <a name="option-c1"></a>
+
+Add a unique endpoint for sending a simple HTTP request with only the SDK version and a bit of meta-data, such as the DSN, to notify Sentry about failed SDKs. We must keep this logic as simple as possible, and it should hardly ever change to drastically minimize the risk of causing more damage. The HTTP request must not use other parts of the SDK, such as client, hub, or transport. The SDKs must only send this request once. As we can’t have any logic running, such as rate-limiting or client reports, it’s good to have a specific endpoint for this to reduce the potential impact on the rest of the infrastructure.
+
+#### Pros <a name="option-c1-pros"></a>
+
+1. We know when a SDK disables itself.
+
+#### Cons <a name="option-c1-cons"></a>
+
+1. Potential risk of crashing while performing this action.
+2. It requires extra infrastructure.
+3. We don't know why the SDK disabled itself.
+
+### Option C2: SDK Retry Logic <a name="option-c2"></a>
+
+When the SDK either operates in Safe Mode or NoOp Mode, it retries to go back to normal or safe mode after running x times in Safe Mode or NoOp Mode.
+
+#### Pros <a name="option-c2-pros"></a>
+
+1. We don't fly completely blind when the SDK is in NoOp Mode.
+2. We reduce the risk of falsely staying in the NoOp or Safe Mode.
+
+#### Cons <a name="option-c2-cons"></a>
+
+1. The SDK might cause extra crashes, which could confuse users.
+
+### Option C3: Anomaly Detection <a name="option-c3"></a>
```

Comment:
much to the pros mentioned below, this could be **very** useful as a signal, independently of this RFC and whether it's caused by SDK crashes or other 🔥